### DISCUSSION TITLE
Detach a forked process – signals to a disconnected and unref'ed process

### DISCUSSION THREAD

**Discussion by Nickwiz** [nodejs/help#4719] [Q&A]
I `fork` a child in a master script using:
```js
const proc = child_process.fork(sub_script, [], {
	// out / err: logfiles
	stdio: ['ignore', out, err, 'ipc'],
	detached: true,
});
proc.on('disconnect', ...);
```
The child does a startup routine then do `process.disconnect()`.

The master script receives the `disconnected` event and continue with operations dependent on the spawned child.

Now: If I send a `SIGINT` or `SIGTERM` to the master it cleans up and exits. All OK.

But: If I omit the `detached: true` when forking the child the child process also receives these signals. Even if I do a `.unref()` in addition to `disconnect()`.

My question is more on the curious side as of why the child get this signal even when both detached and dereferenced?

The `proc.subprocess.channel` is also undefined.

**Comment by bnoordhuis** [comment]
That's simply how the UNIX process model works. It's not specific to Node.js.
